jekyll/_cci2/deployment-examples.md

```diff
@@ -0,0 +1,685 @@
+      cf set-env <app-name-dark> circle_build_num ${CIRCLE_BUILD_NUM}
+      cf set-env <app-name-dark> circle_commit ${CIRCLE_SHA1}
+      cf set-env <app-name-dark> circle_workflow_guid ${CIRCLE_WORKFLOW_ID}
+      cf set-env <app-name-dark> circle_user ${CIRCLE_PROJECT_USERNAME}
+      cf set-env <app-name-dark> circle_repo ${CIRCLE_PROJECT_REPONAME}
+      # Start the application
+      cf start <app-name-dark>
+      # Ensure dark route is exclusive to dark app
+      cf unmap-route <app-name> <example.com> -n dark || echo "Dark Route Already exclusive"
+```
+
+#### Live Deployment
+{:.no_toc}
+
+Until now, the previously pushed "app-name" has not changed.  The final step is to route the production URL to our dark application, stop traffic to the previous version, and rename the applications.
+
+```
+- run:
+    name: Re-route live Domain to latest
+    command: |
+      # Send "real" url to new version
+      cf map-route app-name-dark example.com -n www
+      # Stop sending traffic to previous version
+      cf unmap-route app-name example.com -n www
+      # stop previous version
+      cf stop app-name
+      # delete previous version
+      cf delete app-name -f
+      # Switch name of "dark" version to claim correct name
+      cf rename app-name-dark app-name
+```
+
+#### Manual Approval
+{:.no_toc}
+
+For additional control or validation, you can add a manual "hold" step between the dark and live steps as shown in the sample workflow below.
+
+```
+workflows:
+  version: 2 # only requires if using `version: 2` config.
+  build-deploy:
+    jobs:
+      - test
+      - dark-deploy:
+          requires:
+            - test
+          filters:
+            branches:
+              only: master
+      - hold:
+          type: approval
+          requires:
+            - dark-deploy
+          filters:
+            branches:
+              only: master
+      - live-deploy:
+          requires:
+            - hold # manual approval required via the CircleCI UI to run the live-deploy job
+          filters:
+            branches:
+              only: master
+```
+
+## Firebase
+
+In order to deploy to Firebase you will need to add `firebase-tools` to your project's devDependencies since attempting to install firebase-tools globally in CircleCI will not work.
+
+```
+npm install --save-dev firebase-tools
+```
+
+Generate a Firebase CLI token using the following command:
+
+```
+firebase login:ci
+```
+
+Add the generated token to the CircleCI project's environment variables as `$FIREBASE_DEPLOY_TOKEN`. {% include snippets/env-var-or-context.md %}
+
+The following example shows how you can add a deploy to Firebase job to your project's `config.yml` file. This snippet assumes you already have a job to build your application, called `build-job`, and introduces a deployment workflow that only runs the deployment job once the build job has completed **and** you're on the master branch.
+
+```
+
+ deploy-job:
```

Comment:
```suggestion
  deploy-job:
```
Should be two spaces of indentation for correct yaml.

Reply:
Actually, I think everything in this block might need one additional space. also, this example is pretty close to being a full config, no? Maybe we should just add the full thing for increased copy-paste-ability (I think it just needs a `version` and `jobs` key

Reply:
Yeah that's a good idea, I'll see what I can do

Reply:
I'm going to add this to my to do list - I think Adam is working on a new Firebase example anyway.